### DISCUSSION TITLE
Issue with Exporting Field Results Using PyAEDT

### DISCUSSION THREAD

**Discussion by chamanth-vct** [ansys/pyaedt#5182] [Q&A]
Hi 

I am encountering an issue when trying to export field results into a file using the PyAEDT package. The following error message is displayed:

`PyAEDT ERROR: case file format is not supported for this plot.`

I was able to reproduce the issue with the following code and the sample file ([link to download](https://apps.vcollab.com/fileshare/downloadsharedfile?file_ID=3ob5k4m193cbmz)).

```py
import pyaedt

file_path = r"C:\Users\chamanth\testing_error\Project3.aedt"
exports_folder_path = r"C:\Users\chamanth\testing_error\exports_folder"

desktop = pyaedt.Desktop(non_graphical=True) 
prj= desktop.load_project(file_path)
model = prj.modeler
part_names = model.object_names

post = prj.post  # post_processor

#Exporting Result

for name in part_names:

    h_field_plot = post.create_fieldplot_volume([name],quantity="Vector_H",plot_name="H_vector_plot")

    export_status = post.export_field_plot(plot_name = "H_vector_plot",
                            output_dir = exports_folder_path,
                            file_name = name,
                            file_format = 'case'
                                        )

   #if export_status:
   #   print(f"Exported H_Vector for {name}")
   # else:
   #    print(f"Not Exported for {name}")

    h_field_plot.delete()

desktop.close_desktop()

```
In this code, I am attempting to export the H vector field for each object (part). There are 10 parts in total, and the code works fine for 4 of them. However, for the remaining 6 parts, the above error is thrown as shown in the log below:

```txt
PyAEDT INFO: Modeler class has been initialized! Elapsed time: 0m 0sec
PyAEDT INFO: Parsing C:/Users/chamanth/Downloads/testing_error/Project3.aedt.
PyAEDT INFO: File C:/Users/chamanth/Downloads/testing_error/Project3.aedt correctly loaded. Elapsed time: 0m 0sec
PyAEDT INFO: aedt file load time 0.1261458396911621
PyAEDT INFO: PostProcessor class has been initialized! Elapsed time: 0m 0sec
PyAEDT INFO: Post class has been initialized! Elapsed time: 0m 0sec
PyAEDT ERROR: case file format is not supported for this plot.
PyAEDT ERROR: case file format is not supported for this plot.
PyAEDT ERROR: case file format is not supported for this plot.
PyAEDT ERROR: case file format is not supported for this plot.
PyAEDT ERROR: case file format is not supported for this plot.
PyAEDT ERROR: case file format is not supported for this plot.
PyAEDT INFO: Desktop has been released and closed.
```

**However, I can successfully export the H-field data for these specific parts using the GUI. The issue only arises when automating the process through PyAEDT.**

Could anyone assist me in resolving this issue? Any insights or suggestions would be greatly appreciated.

Thank you.

**Comment by Samuelopez-ansys** [comment]
Hi @chamanth-vct ,
When it works for you in the UI, in which format are you exporting .case or another one?
Case format is not compatible with all AEDT applications, and for instance, in HFSS, it is only compatible with First order elements.

**Comment by chamanth-vct** [comment]
Hi @Samuelopez-ansys,

Thank you for your response.

From the GUI, the `.case` format worked without any issues.
Based on your suggestion, I tried changing the `file_format` to `aedtplt` in the code, but I encountered the following error:

`PyAEDT ERROR: aedtplt file format is not supported for this plot.`

Also, thank you for the information regarding the differences between `.case` and `.aedtplt`. Could you please direct me to any web page or documentation that explain these differences in more detail?

Thank you

**Comment by Samuelopez-ansys** [comment]
Which AEDT versiona re you using?

This examples works:
https://aedt.docs.pyansys.com/version/stable/examples/03-Maxwell/Maxwell2D_Electrostatic.html#sphx-glr-examples-03-maxwell-maxwell2d-electrostatic-py

And unit tests are passing, so it has to be something related to your project. Please share it if you do not find a solution.

.case is an Ansys Ensight format, that you can load into Ansys Ensight or PyVista. I think the documentation is wrong, and the available formats are .case and fldplt. Both contains the field values and node position, but with different formats.

**Reply by chamanth-vct** [reply]
Hi @Samuelopez-ansys,

I am currently using AEDT version 2024 R1 and PyAEDT version 0.10.2.

Unfortunately, I haven't found a solution to this issue yet. I'm facing the problem with a large number of files, and I don't believe there is an issue with the model itself, as I'm able to export the field results easily through the GUI. The failure occurs only when using the code. 

Could you please test this on the provided sample file with the code? [Download link](https://apps.vcollab.com/fileshare/downloadsharedfile?file_ID=3ob5k4m1aiv56h)

This issue is critically impacting our ability to automate the export of field results using PyAEDT.

**Comment by Samuelopez-ansys** [comment]
@chamanth-vct You should first run it in graphical mode, I saw that the plot was not correct:
![image](https://github.com/user-attachments/assets/576ee3e6-2c98-4885-b5d1-8fe9914d84ff)


I think there is a bug, but I need further investigation.

**Comment by chamanth-vct** [comment]
Hi @Samuelopez-ansys 

Thank you for acknowledging the bug and starting the investigation.

In the meantime, I’d like to ask if you could provide some insight into the types of models or scenarios where this issue might occur. We’re encountering this error with a large number of files but haven't been able to identify any clear pattern in these occurrences.

If you could help us by outlining potential scenarios or model characteristics that might trigger this issue, it would be a great help.

Thank you again for your support.

**Comment by chamanth-vct** [comment]
This issue is resolved. 
Check out this issue: https://github.com/ansys/pyaedt/issues/5243
